spec/javascripts/components/projects/project_board_spec.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import ProjectBoard from 'components/projects/ProjectBoard';
+import ProjectLoading from 'components/projects/ProjectLoading';
+
+describe('<ProjectBoard />', () => {
+  describe('when projectBoard.isFetched is false',  () => {
+    it('renders <ProjectLoading />', () => {
+      const fakeProjectBoard = { isFetched: false };
+      const wrapper = shallow(<ProjectBoard projectBoard={fakeProjectBoard} />);
+      const spinnerLoading = wrapper.find('[data-id="project-loading"]');
```

Comment:
I think you could move this setup to `beforeEach`. Also, I believe you can check if the component exists 🤔 :
```diff
- expect(spinnerLoading).toBeTruthy();
+ expect(component.contains(<ProjectLoading />)).toBeTruthy();
```

All the test:
```js
import ProjectLoading from 'components/projects/ProjectLoading';

describe('<ProjectBoard />', () => {
  let wrapper;

  describe('when projectBoard.isFetched is false',  () => {
    beforeEach(() => {
      wrapper = shallow(<ProjectBoard projectBoard={{ isFetched: false }} />);
    });

    it('renders <ProjectLoading />', () => {
      expect(component.contains(<ProjectLoading />)).toBeTruthy();
    });
  });

  describe('when projectBoard.isFetched is true', () => {
    beforeEach(() => {
      wrapper = shallow(<ProjectBoard projectBoard={{ isFetched: true }} />);
    });

    it('does not renders <ProjectLoading />', () => {
      expect(wrapper.contains(<ProjectLoading />)).toBeFalsy()
    });
  });
});
```

Reply:
Thanks!!

Reply:
I'd avoid using `contains` passing the component that way. The reason is that the components that are used inside another component are like private methods, we should not address them directly in tests. A safer way to do it is to use data-attributes that decouple what component is used internally so if you ever change the internal implementation of the component but keep the same output you'll use the same data-id there and the test will continue to pass (which should happen because the component, in this case, was only refactored, not broken). You can change this:

```js
expect(wrapper.contains(<ProjectLoading />)).toBeFalsy()
```

with this:

```js
expect(wrapper.find('[data-id="project-loading"]').exists()).toBeFalsy()
```

Also, regarding moving the rendering to the before each: I don't think it's a good idea either.
If you follow the [Arrange-Act-Assert](http://wiki.c2.com/?ArrangeActAssert) methodology for tests, you'll see that the rendering of the component is literally the second part, which shouldn' be moved as a detail or setup of the test. If you want to make your tests more DRY you can create this function inside the most external `define` of your file:

```js
const render = (props) => shallow(<ProjectBoard {...props } />);
```

And then in your tests, you call `render({ projectBoard: { ... })`